lib/enum/basic_enum.rb

```diff
@@ -22,48 +22,62 @@ def number_of_elements_in_collection
   # This one also uses 'count'
   # Try it yourself!
   def number_of_floats_or_fixnums
+    collection.count{ |num| num.is_a? (Numeric)}
   end
 
   # Refer to: all?
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-all-3F
   def all_words_longer_than_length?(min_length)
+    collection.all? { |word| word.length > min_length}
   end
 
 
   # For the next 3 methods refer to: collect or map
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-any-3F
   def contains_a_word_longer_than?(min_length)
+     collection.any? { |word| word.length > min_length} # I don't understand my any? works. Any word?
   end
 
   def capitalize_words
+    collection.map(&:capitalize)
   end
 
   def square_numbers
+    collection.map { |num| num ** 2}
   end
 
   # Use symbol to proc notation on this one!
   def reverse_words
+    collection.map(&:reverse)
   end
 
   # Refer to collect_concat or flat_map for these next two
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-flat_map
   # To get the order right for the tests, you'll need to use the absolute value method
   def positive_and_negative_numbers
+    collection.map { |num| num.abs}.flat_map{ |num| [num, -num]}
+
   end
 
   # Refer to detect or find
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-detect
   def find_first_awesome_person
+    collection.detect{ |name| name[:awesome]}# how does name[:awesome]work?
```

Comment:
The collection is made of people, so I think this might make more sense if you call that local variable `person` instead of `name`. 

A single person is this hash: `{name: "JohnDoe", awesome: false}`

Let's say I call that `person_1`. 

Now I want to see if `person_1` is awesome or not. Since a hash is a bunch of key-value pairs, I can look up the value for the `awesome` key like this: `person_1[:awesome]`. That expression should return `false`. 

I can also look up the name like this: `person_1[:name]`. Remember that we use symbols for hash keys, so `[:name]` not `[name]`. That should return `"JohnDoe"`. 

The detect method tests the condition you give, and the first time it's true (actually, the first time it's not false), that element is the result of the method. So the first element where `person[:awesome]` returns `true`, that's the result.